xbmc/video/dialogs/GUIDialogVideoVersion.cpp

```diff
@@ -0,0 +1,1008 @@
+    return;
+  }
+
+  // play the video item directly
+  if (!item->GetVideoInfoTag()->m_hasVideoVersions)
+  {
+    callback(item);
+    return;
+  }
+
+  // prompt to play selected video version
+  CGUIDialogVideoVersion* dialog =
+      CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogVideoVersion>(
+          WINDOW_DIALOG_VIDEO_VERSION);
+  if (!dialog)
+    return;
+
+  dialog->SetVideoItem(item, true);
+  dialog->SetPlayCallback(PlayVideoItem);
+  dialog->Open();
+}
+
+int CGUIDialogVideoVersion::ManageVideoVersionContextMenu(const std::shared_ptr<CFileItem>& version)
+{
+  CContextButtons buttons;
+
+  buttons.Add(CONTEXT_BUTTON_RENAME, 118);
+  buttons.Add(CONTEXT_BUTTON_SET_DEFAULT, 31614);
+  buttons.Add(CONTEXT_BUTTON_DELETE, 15015);
+  buttons.Add(CONTEXT_BUTTON_SET_ART, 13511);
+
+  int button = CGUIDialogContextMenu::ShowAndGetChoice(buttons);
+  if (button > 0)
+  {
+    CGUIDialogVideoVersion* dialog =
+        CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogVideoVersion>(
+            WINDOW_DIALOG_VIDEO_VERSION);
+    if (!dialog)
+      return -1;
+
+    CFileItem videoItem;
+    if (!dialog->m_database.GetVideoItemByVideoVersion(version->GetVideoInfoTag()->m_iDbId,
+                                                       videoItem))
+      return -1;
+
+    dialog->SetVideoItem(std::make_shared<CFileItem>(videoItem), false);
+    dialog->SetSelectedVideoVersion(version);
+
+    switch (static_cast<CONTEXT_BUTTON>(button))
+    {
+      case CONTEXT_BUTTON_RENAME:
+      {
+        dialog->Rename();
+        break;
+      }
+      case CONTEXT_BUTTON_SET_DEFAULT:
+      {
+        dialog->SetDefault();
+        break;
+      }
+      case CONTEXT_BUTTON_DELETE:
+      {
+        dialog->Remove();
+        break;
+      }
+      case CONTEXT_BUTTON_SET_ART:
+      {
+        dialog->ChooseArt();
+        break;
+      }
+      default:
+        break;
+    }
+  }
+
+  return button;
+}
+
+int CGUIDialogVideoVersion::SelectVideoVersion(const std::shared_ptr<CFileItem>& item)
+{
+  if (!item || !item->HasVideoInfoTag())
+    return -1;
+
+  VideoDbContentType itemType = item->GetVideoContentType();
+  if (itemType != VideoDbContentType::MOVIES)
+    return -1;
+
+  CVideoDatabase videodb;
+  if (!videodb.Open())
+  {
+    CLog::Log(LOGERROR, "{}: Failed to open database", __FUNCTION__);
+    return -1;
+  }
+
+  int idVideoVersion = -1;
+  std::string typeVideoVersion;
+
+  CFileItemList list;
+  videodb.GetVideoVersionTypes(itemType, list);
+
+  std::string mediaType = item->GetVideoInfoTag()->m_type;
+
+  while (true)
+  {
+    CGUIDialogSelect* dialog =
+        CServiceBroker::GetGUI()->GetWindowManager().GetWindow<CGUIDialogSelect>(
+            WINDOW_DIALOG_SELECT);
+    if (!dialog)
+      return -1;
+
+    dialog->Reset();
+    dialog->SetItems(list);
+    dialog->SetHeading(
+        StringUtils::Format(g_localizeStrings.Get(40003), CMediaTypes::GetLocalization(mediaType)));
```

Comment:
You could optimize this a bit, by not only obtaining the media type before entering the while loop, but the whole heading string, as it will not change as well.